compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.cpp

```diff
@@ -571,6 +609,18 @@ FailureOr<Value> MMAAttr::buildMmaOperation(OpBuilder &builder, Location loc,
     return failure();
   }
   switch (getIntrinsic().getValue()) {
+  case MMAIntrinsic::MFMA_F32_16x16x4_F32: {
+    // Update the lhs and rhs to extract the first element since vector<1xT> is
+    // not supoorted by amgpu.mfma op.
+    Value zeroIdx = builder.create<arith::ConstantIndexOp>(loc, 0);
+    lhs = builder.create<vector::ExtractElementOp>(loc, lhs, zeroIdx);
+    rhs = builder.create<vector::ExtractElementOp>(loc, rhs, zeroIdx);
```

Comment:
Use `vector.extract` with an attribute index instead

Reply:
Updated! Any reasons why?

Reply:
extract element allows for dynamism over the index and, in this application, requires you to materialize the index attribute as a constant. we don't need either of two